core/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -568,50 +568,16 @@ private[chisel3] object ir {
   case class DefClass(id: Class, name: String, ports: Seq[Port], block: Block) extends Component
 
   case class Circuit(
-    name:           String,
-    components:     Seq[Component],
-    annotations:    Seq[ChiselAnnotation],
-    renames:        RenameMap,
-    newAnnotations: Seq[ChiselMultiAnnotation],
-    typeAliases:    Seq[DefTypeAlias],
-    layers:         Seq[Layer],
-    options:        Seq[DefOption]
+    name:        String,
+    components:  Seq[Component],
+    annotations: Seq[() => Seq[Annotation]],
+    renames:     RenameMap,
+    typeAliases: Seq[DefTypeAlias],
+    layers:      Seq[Layer],
+    options:     Seq[DefOption]
   ) {
 
-    def this(
-      name:        String,
-      components:  Seq[Component],
-      annotations: Seq[ChiselAnnotation],
-      renames:     RenameMap,
-      typeAliases: Seq[DefTypeAlias],
-      layers:      Seq[Layer],
-      options:     Seq[DefOption]
-    ) =
-      this(name, components, annotations, renames, Seq.empty, typeAliases, layers, options)
-
     def firrtlAnnotations: Iterable[Annotation] =
-      annotations.flatMap(_.toFirrtl.update(renames)) ++ newAnnotations.flatMap(
-        _.toFirrtl.flatMap(_.update(renames))
-      )
-  }
-
-  object Circuit
-      extends scala.runtime.AbstractFunction7[String, Seq[Component], Seq[ChiselAnnotation], RenameMap, Seq[
-        DefTypeAlias
-      ], Seq[Layer], Seq[DefOption], Circuit] {
-    def unapply(c: Circuit): Option[(String, Seq[Component], Seq[ChiselAnnotation], RenameMap, Seq[DefTypeAlias])] = {
-      Some((c.name, c.components, c.annotations, c.renames, c.typeAliases))
-    }
-
-    def apply(
-      name:        String,
-      components:  Seq[Component],
-      annotations: Seq[ChiselAnnotation],
```

Comment:
All of this was left over from previous needs to maintain binary compatibility. These APIs are all package private now so we can just delete them.

Reply:
Backing out this change for now because `Circuit` leaks as discussed in https://github.com/chipsalliance/chisel/pull/4683. Will follow up tearing this out later.